pkg/clients/router/router.go

```diff
@@ -0,0 +1,248 @@
+				for ipIdx, ipRange := range peer.AdvertisedIpRanges {
+					spec.BgpPeers[idx].AdvertisedIpRanges[ipIdx] = &v1alpha1.RouterAdvertisedIpRange{
+						Description: &ipRange.Description,
+						Range:       ipRange.Range,
+					}
+				}
+			}
+		}
+	}
+
+	if len(in.Nats) != 0 && len(spec.Nats) == 0 {
+		spec.Nats = make([]*v1alpha1.RouterNat, len(in.Nats))
+		for idx, nat := range in.Nats {
+			spec.Nats[idx] = &v1alpha1.RouterNat{
+				DrainNatIps:                      nat.DrainNatIps,
+				EnableEndpointIndependentMapping: &nat.EnableEndpointIndependentMapping,
+				IcmpIdleTimeoutSec:               &nat.IcmpIdleTimeoutSec,
+				MinPortsPerVm:                    &nat.MinPortsPerVm,
+				Name:                             &nat.Name,
+				NatIpAllocateOption:              nat.NatIpAllocateOption,
+				NatIps:                           nat.NatIps,
+				SourceSubnetworkIpRangesToNat:    nat.SourceSubnetworkIpRangesToNat,
+				TcpEstablishedIdleTimeoutSec:     &nat.TcpEstablishedIdleTimeoutSec,
+				TcpTransitoryIdleTimeoutSec:      &nat.TcpTransitoryIdleTimeoutSec,
+				UdpIdleTimeoutSec:                &nat.UdpIdleTimeoutSec,
+			}
+			if nat.LogConfig != nil {
+				spec.Nats[idx].LogConfig = &v1alpha1.RouterNatLogConfig{
+					Enable: &nat.LogConfig.Enable,
+					Filter: &nat.LogConfig.Filter,
+				}
+			}
+
+			if nat.Subnetworks != nil {
+				spec.Nats[idx].Subnetworks = make([]*v1alpha1.RouterNatSubnetworkToNat, len(nat.Subnetworks))
+				for subnetIdx, subnet := range nat.Subnetworks {
+					spec.Nats[idx].Subnetworks[subnetIdx] = &v1alpha1.RouterNatSubnetworkToNat{
+						Name:                  &subnet.Name,
+						SecondaryIpRangeNames: subnet.SecondaryIpRangeNames,
+						SourceIpRangesToNat:   subnet.SourceIpRangesToNat,
+					}
+				}
+			}
+		}
+	}
+
+	if len(in.Interfaces) != 0 && len(spec.Interfaces) == 0 {
+		spec.Interfaces = make([]*v1alpha1.RouterInterface, len(in.Interfaces))
+		for idx, routerInterface := range in.Interfaces {
+			spec.Interfaces[idx] = &v1alpha1.RouterInterface{
+				IpRange:                      &routerInterface.IpRange,
+				LinkedInterconnectAttachment: &routerInterface.LinkedInterconnectAttachment,
+				LinkedVpnTunnel:              &routerInterface.LinkedVpnTunnel,
+				Name:                         routerInterface.Name,
+			}
+		}
+	}
+}
+
+// IsUpToDate checks whether current state is up-to-date compared to the given
+// set of parameters.
+func IsUpToDate(name string, in *v1alpha1.RouterParameters, observed *compute.Router) (upTodate bool, err error) {
+	generated, err := copystructure.Copy(observed)
+	if err != nil {
+		return true, errors.Wrap(err, errCheckUpToDate)
+	}
+	desired, ok := generated.(*compute.Router)
+	if !ok {
+		return true, errors.New(errCheckUpToDate)
+	}
+	GenerateRouter(name, *in, desired)
+	return cmp.Equal(desired, observed, cmpopts.EquateEmpty(), cmpopts.IgnoreFields(compute.Router{}, "ForceSendFields")), nil
```

Comment:
```suggestion
	return cmp.Equal(desired, observed, cmpopts.EquateEmpty(), gcp.EquateComputeURLs(), cmpopts.IgnoreFields(compute.Router{}, "ForceSendFields")), nil
```

Without this, `IsUpToDate` always returns `false` with the following diff:
```
  &compute.Router{
        ... // 8 identical fields
        Name:     "router-test",
        Nats:     {&{MinPortsPerVm: 2, Name: "router-nat-1", NatIpAllocateOption: "AUTO_ONLY", SourceSubnetworkIpRangesToNat: "ALL_SUBNETWORKS_ALL_IP_RANGES", ...}},
-       Network:  "projects/wesaas-playground/global/networks/network-example",
+       Network:  "https://www.googleapis.com/compute/v1/projects/wesaas-playground/global/networks/network-example",
        Region:   "https://www.googleapis.com/compute/v1/projects/wesaas-playground"...,
        SelfLink: "https://www.googleapis.com/compute/v1/projects/wesaas-playground"...,
        ... // 1 ignored and 2 identical fields
  }
```

P.S: You can always debug/get this diff with `cmp.Diff` with the same parameter set you used in `cmp.Equal`